pkg/genericapiserver/storage_factory.go

```diff
@@ -255,16 +254,7 @@ func NewStorageCodec(storageMediaType string, ns runtime.StorageSerializer, stor
 	}
 
 	ds := recognizer.NewDecoder(s, ns.UniversalDeserializer())
-	encoder := ns.EncoderForVersion(s, storageVersion)
-	decoder := ns.DecoderToVersion(ds, memoryVersion)
-	if memoryVersion.Group != storageVersion.Group {
-		// Allow this codec to translate between groups.
-		if err := versioning.EnableCrossGroupEncoding(encoder, memoryVersion.Group, storageVersion.Group); err != nil {
-			return nil, fmt.Errorf("error setting up encoder from %v to %v: %v", memoryVersion, storageVersion, err)
-		}
-		if err := versioning.EnableCrossGroupDecoding(decoder, storageVersion.Group, memoryVersion.Group); err != nil {
-			return nil, fmt.Errorf("error setting up decoder from %v to %v: %v", storageVersion, memoryVersion, err)
-		}
-	}
+	encoder := ns.EncoderForVersion(s, runtime.NewMultiGroupVersioner(storageVersion, storageVersion.Group, memoryVersion.Group))
```

Comment:
Should we start forcing `GroupKind` to be passed in addition to `GroupResource` in the method calls above?  Not necessarily in this pull, but I think it may make the flow a little more obvious.